src/App.tsx

```diff
@@ -1,93 +1,94 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import { TodoApp } from './components/TodoApp';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [activeFilter, setActiveFilter] = useState<Filter>(Filter.All);
+
+  useEffect(() => {
+    const todosFromLocalStorage = localStorage.getItem('todos');
+
+    if (todosFromLocalStorage) {
+      setAllTodos(JSON.parse(todosFromLocalStorage));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(allTodos));
+  }, [allTodos]);
+
+  const handleAddTodo = (title: string) => {
+    const newTodo: Todo = {
+      id: +new Date(),
```

Comment:
Date is not the best id
Use a more reliable method to generate a unique identifier, such as 'uuid'.

Reply:
still not fixed